src/groups/meta_group_wrapper.hpp

```diff
@@ -25,43 +26,61 @@ struct toJs_impl<member> {
-        } else if (info.invite_status == 0) {
-            obj["memberStatus"] = toJs(env, "INVITE_ACCEPTED");
-        } else if (info.invite_not_sent()) {
-            obj["memberStatus"] = toJs(env, "INVITE_NOT_SENT");
-        } else if (info.invite_failed()) {
-            obj["memberStatus"] = toJs(env, "INVITE_FAILED");
-        } else if (info.invite_pending()) {
-            // Note: INVITE_NOT_SENT is 3, which makes invite_pending() return true, so be sure to
-            // check for invite_not_sent() above.
-            obj["memberStatus"] = toJs(env, "INVITE_SENT");
-        } else {
-            obj["memberStatus"] = toJs(env, "UNKNOWN");
+
+        auto status = info.status();
+
+        switch (status) {
+            // invite statuses
+            case session::config::groups::member::Status::invite_unknown:
+                obj["memberStatus"] = toJs(env, "INVITE_UNKNOWN");
+                break;
+            case session::config::groups::member::Status::invite_not_sent:
+                obj["memberStatus"] = toJs(env, "INVITE_NOT_SENT");
+                break;
+            case session::config::groups::member::Status::invite_failed:
+                obj["memberStatus"] = toJs(env, "INVITE_FAILED");
+                break;
+            case session::config::groups::member::Status::invite_sent:
+                obj["memberStatus"] = toJs(env, "INVITE_SENT");
+                break;
+            case session::config::groups::member::Status::invite_accepted:
+                obj["memberStatus"] = toJs(env, "INVITE_ACCEPTED");
+                break;
+
+            // promotion statuses
+            case session::config::groups::member::Status::promotion_unknown:
+                obj["memberStatus"] = toJs(env, "PROMOTION_UNKNOWN");
+                break;
+            case session::config::groups::member::Status::promotion_not_sent:
+                obj["memberStatus"] = toJs(env, "PROMOTION_NOT_SENT");
+                break;
+            case session::config::groups::member::Status::promotion_failed:
+                obj["memberStatus"] = toJs(env, "PROMOTION_FAILED");
+                break;
+            case session::config::groups::member::Status::promotion_sent:
+                obj["memberStatus"] = toJs(env, "PROMOTION_SENT");
+                break;
+            case session::config::groups::member::Status::promotion_accepted:
+                obj["memberStatus"] = toJs(env, "PROMOTION_ACCEPTED");
+                break;
+
+            // removed statuses
+            case session::config::groups::member::Status::removed_unknown:
+                obj["memberStatus"] = toJs(env, "REMOVED_UNKNOWN");
+                break;
+            case session::config::groups::member::Status::removed:
+                obj["memberStatus"] = toJs(env, "REMOVED_MEMBER");
+                break;
+            case session::config::groups::member::Status::removed_including_messages:
```

Comment:
We don't need to change this now but these references to variables are pretty damn long. Is there no way to make them shorter via some different importing syntax? Just asking out of interest.

Reply:
done